app/controllers/repos_controller.rb

```diff
@@ -43,25 +49,35 @@ def show
           columns = chart_builder.build_columns do |benchmark_run|
             environment = YAML.load(benchmark_run.environment)
 
-            if environment.is_a?(Hash)
-              temp = ""
-
-              environment.each do |key, value|
-                temp << "#{key}: #{value}<br>"
-              end
+            commit = benchmark_run.initiator
 
-              environment = temp
+            # generate the version object
+            config = {
+              commit: commit.sha1[0..6],
+              commit_date: commit.created_at,
+              commit_message: commit.message.truncate(30)
+            }
+            if environment.is_a?(Hash)
+              config.merge!(environment)
+              # solely for the purpose of generating the correct HTML
+              environment = hash_to_html(environment)
```

Comment:
good point, do you have a preference for whether we should set `config[:environment]` or just add a comment explaining what's going on, if not I'll just add the comment